api/agent/agent.go

```diff
@@ -782,7 +782,21 @@ func (a *agent) runHot(ctx context.Context, call *call, tok ResourceToken, state
 	// WARNING: we wait forever.
 	defer cookie.Close(common.BackgroundContext(ctx))
 
-	err = a.driver.PrepareCookie(ctx, cookie)
+	shouldPull, err := cookie.ValidateImage(ctx)
```

Comment:
I will add timeouts and better error handling with future PRs here. Agent.go changes in this PR as very basic and intentionally kept bare minimum.